examples/api/src/app/api/erc-20/route.ts

```diff
@@ -0,0 +1,78 @@
+import { NextRequest } from "next/server";
+import {
+  getFollowingHolderInfo,
+  getPriceData,
+  parseTokenParam,
+  getTokenInfo,
+} from "./lib/utils";
+
+export async function GET(request: NextRequest) {
+  const fid = request.nextUrl.searchParams.get("fid");
+  const token = request.nextUrl.searchParams.get("token")?.toLowerCase();
+  let tokenAddress = request.nextUrl.searchParams
+    .get("tokenAddress")
+    ?.toLowerCase();
+  let blockchain = request.nextUrl.searchParams
+    .get("blockchain")
+    ?.toLowerCase();
+  const methodName = request.nextUrl.searchParams
+    .get("function")
+    ?.toLowerCase();
+
+  if (token) {
+    const parsedToken = parseTokenParam(token);
+    tokenAddress = parsedToken.tokenAddress;
+    blockchain = parsedToken.blockchain;
+  }
+
+  const { fn, options } = {
+    holders: {
+      fn: getFollowingHolderInfo,
+      options: {
+        headers: new Headers({
+          // Cache for 1 day
+          "Cache-Control": "public, max-age=86400, immutable",
+        }),
+      },
+    },
+    price: {
+      fn: getPriceData,
+      options: {
+        headers: new Headers({
+          // Cache for 1 hour
+          "Cache-Control": "public, max-age=3600, immutable",
+        }),
+      },
+    },
+    token: {
+      fn: getTokenInfo,
+      options: {
+        headers: new Headers({
+          // Cache for 1 month
+          "Cache-Control": "public, max-age=2592000, immutable",
+        }),
+      },
+    },
+  }[methodName];
+
+  if (!fn) {
+    return Response.json({
+      error: "Invalid function",
+    });
+  }
+
+  const result = await fn({
```

Comment:
this is returning HTML  - probably an error in HTML not json, which is causing the endpoint to 500